src/main/java/service/operation/SupplyOperation.java

```diff
@@ -0,0 +1,23 @@
+package service.operation;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+
+public class SupplyOperation implements OperationHandler {
+    private FruitDao fruitDao;
+
+    public SupplyOperation() {
+        this.fruitDao = new FruitDaoImpl();
+    }
```

Comment:
The constructor of `SupplyOperation` should not create a new instance of `FruitDaoImpl`. Instead, it should accept a `FruitDao` instance as a parameter to follow the Dependency Inversion Principle, which is part of SOLID principles.